### PR TITLE
feat(text-field): Support for types- color, date, datetime-local, etc

### PR DESCRIPTION
**Pull request by namannehra**
Label will always float for the following types as they show some UI even if value is empty
- color
- date
- datetime-local
- month
- range
- time
- week

Added `getType` and `setType` methods to foundation. `setType` must be used to update type after initialisation.

### CODE REVIEW

**Comment by namannehra** [comment]
Is this fine? Do I need fix something or make any changes?

File: packages/mdc-textfield/foundation.js [hunk 1]
```diff
@@ -49,9 +48,15 @@ class MDCTextFieldFoundation extends MDCFoundation {
     return !this.isValid() && !this.isFocused_;
   }
 
+  /** @return {boolean} */
```
Comment (williamernest) [inline packages/mdc-textfield/foundation.js]: Change to `@private` and update the name to `shouldAlwaysFloat_`. 
```
  /**
   * @return {boolean}
   * @private
   */
  get shouldAlwaysFloat_() {
    const type = this.getNativeInput_().type;
    return ALWAYS_FLOAT_TYPES.indexOf(type) >= 0;
  }

  /** @return {boolean} */
  get shouldFloat() {
    return this.shouldAlwaysFloat_ || this.isFocused_ || !!this.getValue() || this.isBadInput_();
  }
```

File: packages/mdc-textfield/foundation.js [hunk 2]
```diff
@@ -250,21 +253,40 @@ class MDCTextFieldFoundation extends MDCFoundation {
...
+   * @return {string} The type of the input Element.
+   */
+  getType() {
+    return this.getNativeInput_().type;
+  }
+
+  /**
+   * @param {string} type The type to set on the input Element.
+   */
+  setType(type) {
```
Comment (williamernest) [inline packages/mdc-textfield/foundation.js]: Dynamically changing type isn't a common use case. This should be removed.

File: packages/mdc-textfield/foundation.js [hunk 3]
```diff
@@ -250,21 +253,40 @@ class MDCTextFieldFoundation extends MDCFoundation {
...
-    if (shouldRemoveLabelFloat) {
+    if (!this.shouldFloat) {
       this.receivedUserInput_ = false;
     }
   }
 
+  /**
+   * @return {string} The type of the input Element.
+   */
+  getType() {
```
Comment (williamernest) [inline packages/mdc-textfield/foundation.js]: This can be removed.

**Comment by williamernest** [comment]
@namannehra Sorry for the delay. Thanks for fixing this for us!

**Pull request merged**